### DISCUSSION TITLE
effect persistence

### DISCUSSION THREAD

**Discussion by prschguy1** [PlummersSoftwareLLC/NightDriverStrip#439] [General]
I am working on a new chip config for my Atom light project. Seems it is a bit glitchy though. Trying to see what effects will work on it and which ones will not. Wanted to see how to keep it from trying to reload the same effect after a crash. For this purpose it would be better for me at this time to start back at the top of the list. I have increased the effect duration so I have time to disable or enable the effects in web browser, but ordering does not seem to be possible there. Any help you can provide would be appreciated.

Thanks, Craig

**Comment by rbergen** [comment]
With the updates that were merged into main yesterday, there are two ways to do this:

- If WiFi and the on-chip webserver are working, changing the `rememberCurrentEffect` device config setting. Using `curl` this can be done with the following command:
  
  ```shell
  curl -d "rememberCurrentEffect=0" -X POST http://<device_IP>/settings
  ```

  And then restart the chip. This will make the chip load the effect set from JSON, but always start at the first one.
  
  This way has been available for a number of months now, so should also work if you're working with a slightly older version of the main tree.

- If WiFi and/or the on-chip webserver are not running, you can pass the following build flag in platformio.ini to ignore the effect data that's persisted to JSON altogether: `-DEFFECT_SET_VERSION=0`
  There's documentation about this define available in effects.cpp, at the top of the `LoadEffectFactories()` function.
  
  This way is the one that's been added with yesterday's merge. 

I hope this helps!

**Comment by prschguy1** [comment]
Thank you so much for that advice, Rutger. I added that build flag, and it has done the trick. Will definitely make my life a lot better!

Thanks again, Craig

**Reply by rbergen** [reply]
You're very welcome!
